src/main/java/io/cdap/plugin/ariba/source/AribaBatchSource.java

```diff
@@ -0,0 +1,125 @@
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import org.apache.hadoop.io.LongWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+/**
+ * Batch Source that reads data from an Ariba system.
+ * <p>
+ * LongWritable is the first parameter because that is the key used by Hadoop's
+ * {@code TextInputFormat}. Similarly, StructuredRecord is the second parameter because that
+ * is the value used by Hadoop's {@code StructuredRecord}.
+ * {@code StructuredRecord} is the third parameter because that is what the
+ * source will output. All the plugins included with Hydrator operate on
+ * StructuredRecord.
+ */
+
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(AribaBatchSource.NAME)
+@Description("Reads data from Sap Ariba.")
+public class AribaBatchSource extends BatchSource<LongWritable, StructuredRecord, StructuredRecord> {
+
+  public static final String NAME = "Ariba";
+  private static final Logger LOG = LoggerFactory.getLogger(AribaBatchSource.class);
+  private final AribaPluginConfig pluginConfig;
+  private final AribaServices aribaServices;
+  public String accessToken;
+
+  public AribaBatchSource(AribaPluginConfig pluginConfig) {
+    this.pluginConfig = pluginConfig;
+    aribaServices = new AribaServices(pluginConfig);
+
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    LOG.trace("Initiating Ariba batch source configure pipeline");
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    pluginConfig.validatePluginParameters(failureCollector);
+
+    if (pluginConfig.isSchemaBuildRequired()) {
+      try {
+        accessToken = aribaServices.getAccessToken();
+        // Get metadata if connection is successful
+        if (accessToken != null) {
+          pipelineConfigurer.getStageConfigurer().setOutputSchema(
+            aribaServices.buildOutputSchema(accessToken));
+        }
+      } catch (IOException exception) {
+        failureCollector.addFailure(exception.getMessage(), null);
+        failureCollector.getOrThrowException();
+      } catch (AribaException exception) {
+        attachFieldWithError(exception, failureCollector);
+        failureCollector.getOrThrowException();
+      }
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(null);
+    }
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    // TODO: https://cdap.atlassian.net/browse/PLUGIN-1172
+  }
+
+
+  /**
+   * Checks and attaches the UI fields with its relevant error message.
+   *
+   * @param aribaException   {@code AribaException}
+   * @param failureCollector {@code FailureCollector}
+   */
+  private void attachFieldWithError(AribaException aribaException, FailureCollector failureCollector) {
+
+    String errMsg = AribaUtil.buildAribaServiceError(aribaException);
```

Comment:
move this inside case HttpURLConnection.HTTP_UNAUTHORIZED:

Reply:
Fixed